scrapy/crawler.py

```diff
@@ -124,14 +124,21 @@ def __init__(
 
         self.settings.freeze()
         self.crawling: bool = False
+        self._started: bool = False
         self.spider: Optional[Spider] = None
         self.engine: Optional[ExecutionEngine] = None
 
     @inlineCallbacks
     def crawl(self, *args: Any, **kwargs: Any) -> Generator[Deferred, Any, None]:
         if self.crawling:
             raise RuntimeError("Crawling already taking place")
-        self.crawling = True
+        if self._started:
+            warnings.warn(
+                "Running Crawler.crawl() more than once is deprecated.",
+                ScrapyDeprecationWarning,
+                stacklevel=2,
```

Comment:
With 2 it's still not helpful as it points to twisted/internet/defer.py (as inlineCallbacks turns into several frames), but I'm not sure if it's possible to set a correct number here.